pkg/trace/obfuscate/sql.go

```diff
@@ -104,11 +106,39 @@ func (f *replaceFilter) Filter(token, lastToken TokenKind, buffer []byte) (token
 	switch token {
 	case String, Number, Null, Variable, PreparedStatement, BooleanLiteral, EscapeSequence:
 		return FilteredGroupable, []byte("?"), nil
+	case '?':
+		// Cases like 'ARRAY [ ?, ? ]' should be collapsed into 'ARRAY [ ? ]'
+		return FilteredGroupable, []byte("?"), nil
+	case Table:
+		if f.normalizeTables {
+			return token, f.replaceDigits(buffer), nil
+		}
+		fallthrough
 	default:
 		return token, buffer, nil
 	}
 }
 
+// replaceDigits replaces consecutive sequences of digits with '?',
+// example: "jobs_2020_1597876964" --> "jobs_?_?"
+func (f *replaceFilter) replaceDigits(buffer []byte) []byte {
```

Comment:
This can be a function of its own, no need for it to be a method since it uses nothing on `f` right?

Reply:
I moved it to `obfuscate.go` instead.